pkg/dependencies/factory.go

```diff
@@ -1087,3 +1092,11 @@ func (f *Factory) WithVSphereDefaulter() *Factory {
 
 	return f
 }
+
+func getProxyConfiguration(clusterSpec *cluster.Spec) (httpProxy, httpsProxy string, noProxy []string) {
+	proxyConfiguration := clusterSpec.Cluster.Spec.ProxyConfiguration
+	if proxyConfiguration != nil {
+		return proxyConfiguration.HttpProxy, proxyConfiguration.HttpsProxy, proxyConfiguration.NoProxy
+	}
+	return "", "", []string{}
```

Comment:
you can return nil for the slice

Reply:
Fine for now, but just the kind of convenience function I'd want to have with clusterSpec as the receiver